src/store/watchList/reducer.js

```diff
@@ -0,0 +1,69 @@
+import { concat, without } from 'lodash'
+
+import {
+  REQUEST_MOVIES_START,
+  REQUEST_MOVIES_SUCCESS,
+  REQUEST_MOVIES_ERROR,
+  SET_REMOVE_MOVIE_ERROR,
+  REMOVE_MOVIE_FROM_STORE,
+  CLEANUP_REMOVE_MOVIE_ERRORS
+} from './types'
+
+const reducer = (state = {}, action) => {
+  const { type } = action
+  switch (type) {
+    case REQUEST_MOVIES_START:
+      return {
+        ...state,
+        meta: {
+          loading: true,
+          error: null
+        }
+      }
+    case REQUEST_MOVIES_SUCCESS:
+      return {
+        ...state,
+        meta: {
+          loading: false,
+          currentPage: action.payload.currentPage,
+          totalPages: action.payload.totalPages,
+          error: null
+        },
+        entries: action.payload.entries
+      }
+    case REQUEST_MOVIES_ERROR:
+      return {
+        ...state,
+        meta: {
+          error: action.error,
+          loading: false
+        },
+        entries: null
+      }
+    case REMOVE_MOVIE_FROM_STORE:
+      return {
+        ...state,
+        entries: without(state.entries, action.movieId)
+      }
+    case SET_REMOVE_MOVIE_ERROR:
+      return {
+        ...state,
+        meta: {
+          ...state.meta,
+          removeErrors: concat(state.meta.removeErrors || [], [action.movieId])
```

Comment:
```state.meta.removeErrors?.push(action.movieId)```

Reply:
@VladWithCookies because `removeErrors` can be undefined and in this case I need `[]`

Reply:
try to use optional chaining like on my example above. i think this should work )

Reply:
@VladWithCookies https://codepen.io/y_kosovych/pen/gOLmEWL If I not mistaken It doesn’t help

Reply:
ok, you're right. let's omit `[]` in 2nd argument in `concat` then.

Reply:
@VladWithCookies but it's something like default value, I can't omit it

Reply:
i meaned this: `concat(state.meta.removeErrors || [], action.movieId)`

Reply:
✅